src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -196,50 +196,106 @@ HRESULT CProfilerManager::AddRawProfilerHook(
 
     unique_ptr<CProfilerCallbackHolder> profilerCallbackHolder(new CProfilerCallbackHolder);
 
-    CComPtr<ICorProfilerCallback> pCorProfilerCallback;
-    IfFailRet(pUnkProfilerCallback->QueryInterface(__uuidof(ICorProfilerCallback), (LPVOID*)&pCorProfilerCallback));
-    profilerCallbackHolder->m_CorProfilerCallback = pCorProfilerCallback;
+    // Follow the implementation set by the CLR at
```

Comment:
Can you add additional comments regarding the fact that ICorProfiler implementations may only implement the highest interface?

Reply:
spoke offline, I'll add more details around this. Due to interface inheritance, raw profilers QI can just return the highest supported version but would still contain implementation for all lower versioned.

---
In reply to: [491181036](https://github.com/microsoft/CLRInstrumentationEngine/pull/292#discussion_r491181036) [](ancestors = 491181036)